binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -49,118 +48,126 @@ abstract class SerdeResolverUtils {
+			return new JsonSerde<>(genericRawClazz);
 		}
-		else if (Short.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Short();
+
+		// Finally, just resort to using the fallback
+		return fallbackSerde;
+	}
+
+	private static Serde<?> getStandardDefaultSerde(Class<?> genericRawClazz) {
+		if (Integer.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Integer();
 		}
-		else if (Double.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Double();
+		if (Long.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Long();
 		}
-		else if (Float.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Float();
+		if (Short.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Short();
 		}
-		else if (byte[].class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.ByteArray();
+		if (Double.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Double();
 		}
-		else if (String.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.String();
+		if (Float.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Float();
 		}
-		else if (UUID.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.UUID();
+		if (byte[].class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.ByteArray();
 		}
-		else if (!isSerdeFromStandardDefaults(fallbackSerde)) {
-			// User purposely set a default serde that is not one of the above
-			serde = fallbackSerde;
+		if (String.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.String();
 		}
-		else {
-			// If the type is Object, then skip assigning the JsonSerde and let the fallback mechanism takes precedence.
-			if (!genericRawClazz.isAssignableFrom((Object.class))) {
-				serde = new JsonSerde(genericRawClazz);
-			}
+		if (UUID.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.UUID();
 		}
-		return serde;
+		return null;
 	}
 
 	private static boolean isSerdeFromStandardDefaults(Serde<?> serde) {
-		if (serde != null) {
-			if (Number.class.isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.ByteArray().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.String().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.UUID().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-		}
-		return false;
+		if (serde == null) {
+			return false;
+		}
+		Class<?> serdeClass = serde.getClass();
+		return Serdes.Integer().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Long().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Short().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Double().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Float().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.ByteArray().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.String().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.UUID().getClass().isAssignableFrom(serdeClass);
```

Comment:
This will fall under another SonarCube rule - long boolean expression, but I don't find it very reasonable 😄 

On the other thought I wonder why plain `serde instance Serdes.Integer()` doesn't work instead? Or even `equals()`...
Just because those `Serdes` classes are all `final`, therefore cannot be extension or so:
```
static public final class IntegerSerde extends WrapperSerde<Integer> {
```

Reply:
Yeh, I had the same thoughts about isAssignable here. 

Either way though, I do prefer the concise "long boolean expression" way of doing this as it keeps it easier to read.